src/Filter/AbstractDateFilter.php

```diff
@@ -142,9 +154,20 @@ abstract protected function applyTypeIsEqual(ProxyQueryInterface $query, string
      *
      * @return void
      */
-    protected function applyType(ProxyQueryInterface $query, $operation, $field, ?\DateTime $datetime = null)
+    protected function applyType(BaseProxyQueryInterface $query, $operation, $field, ?\DateTime $datetime = null)
     {
-        $query->field($field)->$operation($datetime);
+        /* NEXT_MAJOR: Remove this deprecation and update the typehint */
+        if (!$query instanceof ProxyQueryInterface) {
+            @trigger_error(sprintf(
+                'Passing %s as argument 1 to %s() is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x'
+                .' and will throw a \TypeError error in version 4.0. You MUST pass an instance of %s instead.',
+                \get_class($query),
+                __METHOD__,
+                ProxyQueryInterface::class
+            ));
+        }
+
+        $query->getQueryBuilder()->field($field)->$operation($datetime);
```

Comment:
can we actually guarantee that the method `getQueryBuilder` exists? I guess not?

Reply:
Before, the `field` method wasnt sure to exist on `$query` too.

Code was written as if it was a ProxyQuery, so I'm pretty sure it's only used with ProxyQuery.
We decided to do this change in DoctrineORM.